### PR TITLE
Clarify Spring Boot starter page

### LINKED ISSUE #3911: Move the Spring starter page documentation to a new page
**What needs to be changed?** Describe the update that is required.

See https://github.com/open-telemetry/opentelemetry.io/pull/3900#issuecomment-1918850387

The Spring Boot page could describe the benefits of the OTel agent and the Spring starter. The documentation of the Spring starter could be moved to another page, referenced by the Spring Boot page.

**What is the name + path of the page that needs changed?**

docs/languages/java/automatic/spring-boot/

### PR DESCRIPTION
**Pull request by jeanbisutti**
Clarify the Spring Boot page and resolves #3911.

 #3911 has proposed to move the Spring Boot starter content to a dedicated page. However, this would add another menu entry and would complexify the documentation discoverability.

So this PR clarify the  Spring Boot page without adding another page:
* It changes the page title from _Spring Boot Starter_ to _Spring Boot_
* It adds an header  _How to instrument Spring Boot with OpenTelemetry?_
* Update the header levels
* Mention the limits if the starter automatic instrumentation compared the Java agent automatic instrumentation

Preview: https://deploy-preview-3985--opentelemetry.netlify.app/docs/languages/java/automatic/spring-boot/

### CODE REVIEW

File: content/en/docs/languages/java/automatic/spring-boot.md [hunk 1]
```diff
@@ -469,12 +480,12 @@ public class TracedClass {
...
 ```
 
-### OpenTelemetry instrumentations libraries
+#### OpenTelemetry instrumentations libraries
 
 You can configure other instrumentations with
 [OpenTelemetry instrumentations libraries](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks).
 
-## Other configurations
+## Other Spring autoconfiguration-based instrumentation solutions
```
Comment (zeitlinger) [inline content/en/docs/languages/java/automatic/spring-boot.md:488]: is this micrometer tracing, which is part of spring boot?

File: content/en/docs/languages/java/automatic/spring-boot.md [hunk 2]
```diff
@@ -26,15 +28,24 @@ following cases:
 [Spring Boot starter]:
   https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.build-systems.starters
 
-The
+The OpenTelemetry Java agent has more automatic instrumentation features than
+the OpenTelemetry starter.
+
+You can use
```
Comment (zeitlinger) [inline content/en/docs/languages/java/automatic/spring-boot.md:34]: maybe add this to the end (other solutions)

File: content/en/docs/languages/java/automatic/spring-boot.md (hunk 1 above)
Comment (zeitlinger) [inline content/en/docs/languages/java/automatic/spring-boot.md:488]:
```suggestion
## Other solutions
```

File: content/en/docs/languages/java/automatic/spring-boot.md [hunk 3]
```diff
@@ -26,15 +28,24 @@ following cases:
...
+[OpenTelemetry instrumentations libraries](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks)
+to complete the automatic instrumentation of the Spring Boot starter.
+
+## OpenTelemetry starter Spring Boot starter
+
+### Spring Boot application type
```
Comment (zeitlinger) [inline content/en/docs/languages/java/automatic/spring-boot.md:40]:
```suggestion
### Compatibility
```

File: content/en/docs/languages/java/automatic/spring-boot.md (hunk 1 above)
Comment (jeanbisutti) [reply content/en/docs/languages/java/automatic/spring-boot.md:488]: OTel Spring autoconfiguration and Zipkin starter

File: content/en/docs/languages/java/automatic/spring-boot.md (hunk 2 above)
Comment (jeanbisutti) [reply content/en/docs/languages/java/automatic/spring-boot.md:34]: I would prefer to mention it at the beginning. In this way, the user could know the advantages and limitations of the OTel Java agent and the OTel Spring starter.

File: content/en/docs/languages/java/automatic/spring-boot.md [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
-title: Spring Boot Starter
+title: Spring Boot
 linkTitle: Spring Boot
 weight: 30
 description: Spring Boot instrumentation for OpenTelemetry Java
 # prettier-ignore
 cSpell:ignore: autoconfigurations autoconfigures datasource logback springboot webflux webmvc
 ---
 
+## How to instrument Spring Boot with OpenTelemetry?
```
Comment (cartermp) [inline content/en/docs/languages/java/automatic/spring-boot.md:10]:
```suggestion
## How to instrument Spring Boot with OpenTelemetry
```

Prefer no questions here

**Review by cartermp — APPROVED** [review]
Overall looks good

**Pull request merged**